test/functional/ui/cmdline_spec.lua

```diff
@@ -857,4 +859,49 @@ describe("cmdline height", function()
     screen:try_resize(25, 5)
     assert_alive()
   end)
+
+  it('cmdheight can be 0', function()
+    clear()
+    local screen = Screen.new(25, 5)
+    screen:attach()
+
+    command("set cmdheight=1 noruler laststatus=2")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      [No Name]                |
+                               |
+    ]]}
+
+    command("set cmdheight=0 noruler laststatus=2")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      ~                        |
+      [No Name]                |
+    ]]}
+
+    command("set cmdheight=0 ruler laststatus=0")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      ~                        |
+      ~                        |
+    ]]}
```

Comment:
should also test what is shown when the user starts typing a `:` command

Reply:
I have added the tests.